sites/fast-creator/app/creator.tsx

```diff
@@ -689,11 +730,39 @@ class Creator extends Editor<{}, CreatorState> {
                 devToolsVisible: !this.state.devToolsVisible,
             },
             () => {
-                this.setViewerToFullSize();
+                const device: Device | void = this.getDeviceById(this.state.deviceId);
+                if (device && device.display === Display.responsive) {
+                    this.setViewerToFullSize();
+                }
                 this.updateMonacoEditor();
             }
         );
     };
+
+    /**
+     * Handle the preview mode switch change event
+     * @param newState - The new state of the switch
+     */
+    private handlePreviewModeSwitch = (newState: boolean): void => {
+        this.setState(
+            {
+                displayMode: newState ? DisplayMode.preview : DisplayMode.default,
+            },
+            () => {
+                // Send message
+                this.fastMessageSystem.postMessage({
+                    type: MessageSystemType.custom,
+                    options: {
+                        originatorId: creatorOriginatorId,
+                        action: newState
+                            ? "displayMode::preview"
```

Comment:
I would store these as const, maybe in the `app/utilities/shared.ts` if both the `creator.tsx` and `preview.tsx` will end up using them.

Reply:
Done